### PR TITLE
Buffer fix

### PR DESCRIPTION
**Pull request by shaunc**
Buffer should be added for both previous and next items, but only previous items were buffered.

### CODE REVIEW

File: addon/components/ember-collection.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default Ember.Component.extend({
     // this.lastCell = undefined;
     // this.cellCount = undefined;
     this.contentElement = undefined;
-    this._cells = [];
+    this._cells = Ember.A([]);
```
Comment (mmun) [inline addon/components/ember-collection.js]: Why did you add this?

**Comment by mmun** [comment]
Is there a way we can test the buffer without digging into state variables on the collection component?

**Comment by mmun** [comment]
Also, please rebase against master. In the ember projects we only merge master for long running branches.

**Comment by shaunc** [comment]
The real problem vs "digging into state" is that I'm doing a calculation that is really done by the layout, which I don't have access to. This could be done if `sliceDidChange` were available much more easily. The # of items in the dom is what I'm checking so I can't depend on DOM to get this info.

**Comment by mmun** [comment]
Gotcha.

**Comment by shaunc** [comment]
I'll rebase ... anything else for the moment? If/when "sliceDidChange" goes through & in order to write tests for `mixedGrid` these may want to be rewritten...

**Comment by mmun** [comment]
No, it's fine how it is other than the Ember.A() thing. You can leave a TODO comment about modifying the tests later if you want.

**Comment by shaunc** [comment]
Ah -- didn't notice your Ember.A comment; sorry:

1) I did get an error earlier in a test that went away when I put Ember.A in init() (I think it was "pushObject not present). Perhaps I still had the package to turn off injections still installed, though it was removed from npm (or bower)?
2) In any case, I thought the guidance for Addons was that we should write without assuming that the ember stuff is injected. If so, the 

      this._cells.pushObject(cell);

further down will break if you don't wrap in Ember.A.

**Comment by mmun** [comment]
> Perhaps I still had the package to turn off injections still installed, though it was removed from npm (or bower)?

That's gotta be it.

`Ember`.A is pretty dubious because it copies all the array extensions onto the array instance. If you want to avoid prototype extensions in your addon, the correct solution is not to copy all the extensions onto the array instance, it is to _not use prototype extensions_ :P

**Comment by mmun** [comment]
I think if you swap `this._cells.pushObject` with `this._cells.push` we can disable all prototype extensions again. We should do that separately though.

**Comment by shaunc** [comment]
Um... so should I put in a separate pull request to not use pushObject?

**Comment by shaunc** [comment]
ah -- just what I was thinking

**Comment by mmun** [comment]
:D I gotta run, I'll be back in a little while.

**Comment by shaunc** [comment]
Sure ... Should I create a new PR with pushObject -> push and https://github.com/rwjblue/ember-disable-prototype-extensions installed?

**Comment by mmun** [comment]
@shaunc Yes exactly. You _might_ have to add Ember.A() but I doubt it.

**Comment by mmun** [comment]
@shaunc and remove this stuff too https://github.com/emberjs/ember-collection/blob/master/tests/dummy/config/environment.js#L9-L20

**Comment by shaunc** [comment]
@mmun -- ok -- I have rebased
